library/psa_crypto_rsa.c

```diff
@@ -62,50 +61,40 @@ psa_status_t mbedtls_psa_rsa_load_representation(
     mbedtls_rsa_context **p_rsa)
 {
     psa_status_t status;
-    mbedtls_pk_context ctx;
     size_t bits;
-    mbedtls_pk_init(&ctx);
+
+    *p_rsa = mbedtls_calloc(1, sizeof(mbedtls_rsa_context));
+    if (*p_rsa == NULL) {
+        return PSA_ERROR_INSUFFICIENT_MEMORY;
+    }
+    mbedtls_rsa_init(*p_rsa);
 
     /* Parse the data. */
     if (PSA_KEY_TYPE_IS_KEY_PAIR(type)) {
-        status = mbedtls_to_psa_error(
-            mbedtls_pk_parse_key(&ctx, data, data_length, NULL, 0,
-                                 mbedtls_psa_get_random, MBEDTLS_PSA_RANDOM_STATE));
+        status = mbedtls_to_psa_error(mbedtls_rsa_key_parse(*p_rsa, data, data_length));
     } else {
-        status = mbedtls_to_psa_error(
-            mbedtls_pk_parse_public_key(&ctx, data, data_length));
+        unsigned char *p = (unsigned char *) data;
+        unsigned char *end = (unsigned char *) (data + data_length);
```

Comment:
I added that cast because I was afraid that some compiler could complain about losing the `const` qualifier in `p` and `end`, but if this is not the case I'm happy to remove it :)

Reply:
Anyway this will be removed after [this change](https://github.com/Mbed-TLS/mbedtls/pull/8740#discussion_r1473430532) so this is addressed somehow :)